python/lsst/ip/isr/linearize.py

```diff
@@ -20,24 +20,21 @@
 # see <http://www.lsstcorp.org/LegalNotices/>.
 #
 import abc
-import copy
-import datetime
-
 import numpy as np
-import yaml
 
-import lsst.afw.table as afwTable
-from lsst.daf.base import PropertyList
+from astropy.table import Table
```

Comment:
Are we moving away from afwTable in favor of astropy.Table?

Reply:
Yes for tables that are all in python and not needed to be in C++. Astropy table is faster than afw table for this use case.

Reply:
@timj, @ktlim — if that's official project policy, can you please make sure it's captured in the Developer Guide? In particular, I think it directly contradicts the statement at https://developer.lsst.io/python/astropy.html that

> If the code is already using afw it is strongly preferred that afw equivalents be used until such time as specific afw interfaces are deprecated

Reply:
It was more a discovery by @czwa that he learned when he was writing these calibration tasks (which means Defects is slow at the moment and can be improved when it becomes a subclass of the new calibration base class). We haven't formally written that discovery down anywhere.

Reply:
That's fine, but I think we do need to make sure that this is captured as part of appropriate policy. “Performance concerns” alone are not an adequate reason to ignore the Dev Guide.

Reply:
I very much endorse the idea of making an exception to this rule for `afw.table`.  But I think this is an easy case to make that call on, and writing a policy to cover all cases (including the role of pandas vs. afw.table for Source/Object catalogs) is a lot harder.  I hope we can capture that action without blocking this ticket on it.

Reply:
I hope we aren't considering blocking this ticket. The performance hit was pretty bad if I recall correctly.

Reply:
This ticket doesn't seem to have anything to do with mitigating a performance hit; I don't really understand why that's coming up here.

However, I don't see any need to block this work, as long as the Architecture Team can reassure us that a) it doesn't actually contravene the Developer Guide, and b) they'll be updating the Developer Guide to provide appropriate guidance in future.

Reply:
This code is bringing this class into compliance with the pre-existing IsrCalib which is the thing that required astropy tables for performance.

Reply:
Ah; it's a shame we didn't take that opportunity to update the Dev Guide! :-)